README.md

```diff
@@ -1,325 +1,293 @@
+
+```sh
+yarn add --dev ts-jest
+```
+
+Then, we'll open up our `package.json` and replace the `jest` field with the following:
+
+```json
+"jest": {
+  "preset": "react-native",
+  "moduleFileExtensions": [
+    "ts",
+    "tsx",
+    "js"
+  ],
+  "transform": {
+    "^.+\\.(js)$": "<rootDir>/node_modules/babel-jest",
+    "\\.(ts|tsx)$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
+  },
+  "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
+  "testPathIgnorePatterns": [
+    "\\.snap$",
+    "<rootDir>/node_modules/"
+  ],
+  "cacheDirectory": ".jest/cache"
+}
+```
+
+This will configure Jest to run `.ts` and `.tsx` files with `ts-jest`.
+
+## Installing Dependency Type Declarations
+
+To get the best experience in TypeScript, we want the type-checker to understand the shape and API of our dependencies. Some libraries will publish their packages with `.d.ts` files (type declaration/type definition files), which can describe the shape of the underlying JavaScript. For other libraries, we'll need to explicitly install the appropriate package in the `@types/` npm scope.
+
+For example, here we'll need types for Jest, React, and React Native, and React Test Renderer.
+
+```ts
+yarn add --dev @types/jest @types/react @types/react-native @types/react-test-renderer
+```
+
+We saved these declaration file packages to our _dev_ dependencies because we're not publishing this package as a library to npm. If we were, we might have to add some of them as regular dependencies.
+
+You can read more [here about getting `.d.ts` files](https://www.typescriptlang.org/docs/handbook/declaration-files/consumption.html).
+
+## Ignoring More Files
+
+For your source control, you'll want to start ignoring the `.jest` folder. If you're using git, we can just add entries to our `.gitignore` file.
+
+```config
+# Jest
+#
+.jest/
+```
+
+As a checkpoint, consider committing your files into version control.
+
+```sh
+git init
+git add .gitignore # import to do this first, to ignore our files
+git add .
+git commit -am "Initial commit."
+```
+
+## Adding a Component
+
+We can now add a component to our app. Let's go ahead and create a `Hello.tsx` component. Create a `components` directory and add the following example.
+
+```ts
+// components/Hello.tsx
+import * as React from "react"
+import { Button, StyleSheet, Text, View } from "react-native"
+
+export interface Props {
+  name: string
+  enthusiasmLevel?: number
+  onIncrement?: () => void
+  onDecrement?: () => void
+}
+
+interface State {
+  enthusiasmLevel: number
+  onIncrement: () => void
+  onDecrement: () => void
+}
+
+export class Hello extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    if ((props.enthusiasmLevel || 0) <= 0) {
+      throw new Error("You could be a little more enthusiastic. :D")
+    }
+
+    this.state = {
```

Comment:
I think that if we're going to use `state`, we should reduce the complexity and just forget about `onIncrement` and `onDecrement` as props, and update state directly. So I'd define

```ts
onIncrement = () => this.setState({ enthusiasmLevel: this.state.enthusiasmLevel + 1 });
onDecrement = () => this.setState({ enthusiasmLevel: this.state.enthusiasmLevel - 1 });
```